### DISCUSSION TITLE
Cannot get it running on Pi4

### DISCUSSION THREAD

**Discussion by JonFonger** [GerryDazoo/Slinger#608] [General]
First off, thank you Gerry for your hard work on Slinger!
I got Slinger working on my Windows computer and everything has been working great.

Rather than running on my PC, I thought I'd try running Slinger on a Raspberry Pi4.

I've followed the instructions on README.TXT and also made sure that the python path is set to the version, python3.

After running these commands:

 sudo systemctl daemon-reload
 sudo systemctl enable sling.service
 sudo systemctl start sling.service

I do not see the log file, sling.log in the tmp folder.
Here is what I see after checking status of sling.service

sudo systemctl status sling.service

● sling.service - SlingBox Server Service
     Loaded: loaded (/etc/systemd/system/sling.service; enabled; vendor preset: enabled)
     Active: failed (Result: exit-code) since Sun 2023-06-25 23:11:04 PDT; 19s ago
    Process: 2150 ExecStart=/usr/bin/env python3 -u slingbox_server.py (code=exited, status=217/USER)
   Main PID: 2150 (code=exited, status=217/USER)
        CPU: 3ms

Jun 25 23:11:04 raspberrypi systemd[1]: sling.service: Scheduled restart job, restart counter is at 5.
Jun 25 23:11:04 raspberrypi systemd[1]: Stopped SlingBox Server Service.
Jun 25 23:11:04 raspberrypi systemd[1]: sling.service: Start request repeated too quickly.
Jun 25 23:11:04 raspberrypi systemd[1]: sling.service: Failed with result 'exit-code'.
Jun 25 23:11:04 raspberrypi systemd[1]: Failed to start SlingBox Server Service.

Seems it didn't run at all...

The config.ini and remote.html is copied from the working Windows Slinger. I'm fairly new to Pi and Linux so pls take it easy on me... (lol). Thanks!

**Comment by GerryDazoo** [comment]
Run it manually to see what's up. 
cd /path_to_server_config
/path_to_python/python3 slingbox_server.py

And report the output.

**Reply by JonFonger** [reply]
Thanks for the reply.  I got it Slinger running manually as suggested.

Changed directory /home/slingbox, where my server_config is stored.
Then, sudo /usr/bin/python3 slingbox_server.py, and Slinger ran perfectly.

So, it seems the problem happens when running sling.service. It is copied to /etc/systemd/system/ as per instructions.

The content of the sling.service file looks like this. 
I suspect it has something to do with this line: ExecStart=/usr/bin/env python3 -u slingbox_server.py ??

[Unit]
Description=SlingBox Server Service
After=systemd-hostnamed.service
Wants=systemd-hostnamed.service

[Service]
Type=idle
User=slingbox
WorkingDirectory=/home/slingbox
StandardOutput=append:/tmp/sling.log
StandardError=append:/tmp/sling.log
ExecStart=/usr/bin/env python3 -u slingbox_server.py 
Restart=always

[Install]
WantedBy=multi-user.target

**Comment by GerryDazoo** [comment]
python3 binary might not be findable by the env command. probably easiest to change the execstart line to /usr/bin/python3.
P.S. You don't need to be root to run the server.

**Comment by JonFonger** [comment]
ok I've changed the line to ExecStart=/usr/bin/python3 -u slingbox_server.py 

then ran these commands to enable it.
 sudo systemctl daemon-reload
 sudo systemctl enable sling.service
 sudo systemctl start sling.service

checked status and looks like this:

● sling.service - SlingBox Server Service
     Loaded: loaded (/etc/systemd/system/sling.service; enabled; vendor preset: enabled)
     Active: failed (Result: exit-code) since Mon 2023-06-26 11:29:40 PDT; 16s ago
    Process: 3558 ExecStart=/usr/bin/python3 -u slingbox_server.py (code=exited, status=217/USER)
   Main PID: 3558 (code=exited, status=217/USER)
        CPU: 3ms

Jun 26 11:29:40 raspberrypi systemd[1]: sling.service: Scheduled restart job, restart counter is at 5.
Jun 26 11:29:40 raspberrypi systemd[1]: Stopped SlingBox Server Service.
Jun 26 11:29:40 raspberrypi systemd[1]: sling.service: Start request repeated too quickly.
Jun 26 11:29:40 raspberrypi systemd[1]: sling.service: Failed with result 'exit-code'.
Jun 26 11:29:40 raspberrypi systemd[1]: Failed to start SlingBox Server Service.

Seems I am close bit fails to run for some reason.

**Reply by skijohn877** [reply]
My PI3b is running python version 3.7.3, my sling.service has "ExecStart=/usr/bin/python3.7 -u slingbox_server.py".

At a command prompt type in "python --version", see what version your running.

**Reply by JonFonger** [reply]
Thanks for the tip skijohn877!

Looks like my python ver is 3.9.2, so I changed the sling service to ExecStart=/usr/bin/python3.9 -u slingbox_server.py  
Unfortunately no dice... I still get the same error status.

● sling.service - SlingBox Server Service
     Loaded: loaded (/etc/systemd/system/sling.service; enabled; vendor preset: enabled)
     Active: failed (Result: exit-code) since Mon 2023-06-26 15:58:17 PDT; 8s ago
    Process: 1570 ExecStart=/usr/bin/python3.9 -u slingbox_server.py (code=exited, status=217/USER)
   Main PID: 1570 (code=exited, status=217/USER)
        CPU: 3ms

Jun 26 15:58:17 raspberrypi systemd[1]: sling.service: Scheduled restart job, restart counter is at 5.
Jun 26 15:58:17 raspberrypi systemd[1]: Stopped SlingBox Server Service.
Jun 26 15:58:17 raspberrypi systemd[1]: sling.service: Start request repeated too quickly.
Jun 26 15:58:17 raspberrypi systemd[1]: sling.service: Failed with result 'exit-code'.
Jun 26 15:58:17 raspberrypi systemd[1]: Failed to start SlingBox Server Service.

I'm going to take a break on this. my head is spinning.. lol
I suppose I can run it manually for now.

**Comment by JonFonger** [comment]
Finally got it working!

Looks like I had to create a user called 'slingbox' in Pi.
Once I did that, I was able to get it running on systemd.

I suppose another way I could have done it was updating the user name in the sling.service file to a user that exists in Pi.
User='my user name'
eg. User=jonfonger

Anyway I hope this helps others that might want to try to get this running on Raspberry Pi.
